app/javascript/utils/lazy-highlight-all.ts

```diff
@@ -0,0 +1,42 @@
+export function lazyHighlightAll(): void {
+  let highlighted = false
+
+  function checkAndHighlight(): void {
+    if (highlighted) return
+
+    // only load highlightjs when a code block with lang-... or language-... classname exists in the DOM
+    if (
+      document.querySelector('code[class^=lang-]') !== null ||
+      document.querySelector('code[class^=language-]') !== null
+    ) {
+      import('@/utils/highlight').then((m) => {
+        m.highlightAll()
+        highlighted = true
```

Comment:
Could there be a race condition where `checkAndHighlight` is being called more than once?

Reply:
It might be called more than once, but once the highlighting is done, it will exit early.
I wonder if there is a case where a code block is loaded statically, we highlight it, then another code block comes dynamically. We wouldn't highlight that, because the function would exit.